ext/symengine/ruby_matrix.c

```diff
@@ -0,0 +1,123 @@
+#include "ruby_matrix.h"
+
+void cmatrix_dense_free(void *ptr)
+{
+    CDenseMatrix *mat_ptr = ptr;
+    dense_matrix_free(mat_ptr);
+}
+
+VALUE cmatrix_dense_alloc(VALUE klass)
+{
+    CDenseMatrix *mat_ptr = dense_matrix_new();
+    return Data_Wrap_Struct(klass, NULL, cmatrix_dense_free, mat_ptr);
```

Comment:
`klass` is the class object. https://github.com/symengine/symengine.rb/blob/master/ext/symengine/ruby_basic.h#L13